### PR TITLE
(application) Add Fade-to-Black Effect, Update Build Scripts

### PR DESCRIPTION
**Pull request by Sayad-Uddin-Tahsin**
- Added a `fade-to-black` effect for a smoother visual transition within the application.
- Updated Windows build scripts to include the `--add-data` flag in the `pyinstaller` command, ensuring correct inclusion of assets directory and icon file path.
- Added `--clean` flag to ensure a clean build process on Windows platform.
- Updated `BlinkEyeApp` class with a `fade_to_black` method for seamless window fading.
- Modified `show_timer_popup` method to utilize the new `fade_to_black` method instead of direct window manipulation.
- Modified application files to a separate directory `/application/*` and necessary icon and image files to a `/application/Assets/*`

Additionally, Address Fedora Package:

- Corrected license file path in `BlinkEye_Fedora.spec`.
- Adjusted `build_fedora.sh` script for Fedora package with specific options like icon addition, data inclusion, hidden import, and clean build.

P.S.: @fluentmoheshwar please review the `Fedora` related modifications!

### CODE REVIEW

**Review by fluentmoheshwar — APPROVED** [review]
LGTM!

**Comment by Sayad-Uddin-Tahsin** [comment]
> LGTM!

Roger that!

File: application/build_fedora.sh
Comment (nomandhoni-cs) [inline application/build_fedora.sh:1]: @Sayad-Uddin-Tahsin , There are some merge conflicts, Kindly resolve them, And I think you've mistakenly pasted the build process for windows 
`pyinstaller --name BlinkEye --onefile --windowed --icon="./Assets/blink-eye-logo.ico" --add-data="./Assets/*;./Assets" --hidden-import plyer.platforms.win.notification --clean blink-eye.py`

**Comment by nomandhoni-cs** [comment]
Kindly resolve these merge conflicts,
image


Also one more thing I want to, inform to both of you ( @Sayad-Uddin-Tahsin , @fluentmoheshwar ), I've created a develop branch, Please make your PR to the develop branch next time. That when our all testing will be done, then we will merge in to master branch

**Comment by fluentmoheshwar** [comment]
@Sayad-Uddin-Tahsin please let me know before solving the merge conflicts. I think it's caused by my latest PR

**Comment by fluentmoheshwar** [comment]
> Kindly resolve these merge conflicts,
> image
> 
> 
> Also one more thing I want to, inform to both of you ( @Sayad-Uddin-Tahsin , @fluentmoheshwar ), I've created a develop branch, Please make your PR to the develop branch next time. That when our all testing will be done, then we will merge in to master branch

Thanks for informing us.

File: application/build_fedora.sh
Comment (Sayad-Uddin-Tahsin) [reply application/build_fedora.sh:1]: @nomandhoni-cs Sorry to bother you, can you please help me to solve this one? It seems to have some other problem... As a Windows guys, I don't have that much knowledge on Linux, maybe you have?

Comment (Sayad-Uddin-Tahsin) [reply application/build_fedora.sh:1]: Does the issue is causing by the `BlinkEye_Fedora.spec`? @fluentmoheshwar

Comment (Sayad-Uddin-Tahsin) [reply application/build_fedora.sh:1]: Or it just simply don't support icon and add-data?

**Comment by Sayad-Uddin-Tahsin** [comment]
@nomandhoni-cs @fluentmoheshwar Reverted all the changes on 
```
BlinkEye_Fedora.spec
build_fedora.sh
```
But I really don't know why it's still showing Conflict?

**Comment by Sayad-Uddin-Tahsin** [comment]
> @Sayad-Uddin-Tahsin please let me know before solving the merge conflicts. I think it's caused by my latest PR

I've tried by reverting all the changes, seems still doesn't work!

**Comment by nomandhoni-cs** [comment]
Can you make the PR to the `develop` branch. Then We will try to fix the issues, instead of force merging the conflict in the master.

**Pull request merged**